synapseclient/core/retry.py

```diff
@@ -91,8 +90,8 @@ def foo(a, b, c): return [a, b, c]
         # Check if we got a retry-able error
         if response is not None and hasattr(response, 'status_code'):
             if (
-                    (retry_status_code_in and response.status_code in retry_status_codes) or
-                    (not retry_status_code_in and response.status_code not in retry_status_codes)
+                (expected_status_codes and response.status_code not in expected_status_codes) or
+                response.status_code in retry_status_codes
```

Comment:
`None` could be passed as value to `retry_status_codes`

Reply:
Added a Falsey check to retry_status_codes before using it